contracts/src/libraries/EnumerableUint64Set.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.20;
+
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+// Followed patterns for EnumerableSet.UintSet
+// See https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol
+library EnumerableUint64Set {
```

Comment:
This may be overkill, and actually use more gas then just adding a list (given double copying of list in values), but wanted to try this out

We could also just write out own enermerable set for chain ids it's not too hard.

Reply:
- reason I didn't import because they didn't have uint64 support. only uint256. we could just wrap the uint256. I also just wanted to try this
- benefit over mapping is that it's enumerable - which we need

though i think given what we need out of our xchain registry, we can just use a list. each function usually requires iterating over each chain anyways